src/macaron/policy_engine/readme.md

```diff
@@ -0,0 +1,91 @@
+Macaron Policy Engine
+-----------------------
+
+This module is used to enforce policy requirements using the output of Macaron's analysis.
+
+It has two modes of operation:
+
+1. When running macaron in `analyze` mode, pass `-po policyfile.dl` and it will enforce the policy on the analyzed
+   repositories.
+2. Independently, by running `macaron.policy_engine`, and passing it a database file from historical analyses.
+
+```
+usage: policy_engine [-h] -d DATABASE [-f FILE] [-s] [-v] [-l LOG_PATH]
+
+options:
+  -h, --help            show this help message and exit
+  -d DATABASE, --database DATABASE
+                        Database path
+  -f FILE, --file FILE  Replace policy file
+  -s, --show-prelude    Show policy prelude
+  -v, --verbose         Enable verbose logging
+  -l LOG_PATH, --log-path LOG_PATH
+                        Log file path
+```
+
+Writing Policies
+----------------
+
+Policies are written using [Souffle datalog](policies.md),
+some example policies can be found [here: ./examples](examples).
+
+To import the facts from the database, or Macaron, policies must begin with the line `#include "prelude.dl"`.
+This imports the automatically generated datalog statements to import the facts.
+
+This includes
+
+1. `.decl`, and `.input` statements to load facts from the database
+2. Rules to derive helper relations from these facts
+3. The relations needed to define Policies so that macaron can understand the results
+4. Some simple pre-written Policies
+
+These can be found in the folder [prelude/](prelude). The automatically generated rules can be found by running
+`macaron.policy_engine -d database.db -s`.
+
+A policy consists of two statements, a definition clause and an enforcement clause. For example, to write a policy that
+requires that the repository has verified authenticated provenance we can write:
+
+```c
+Policy("auth-provenance", repositoryid, "") :-    check_passed(repositoryid, "mcn_provenance_level_three_1").
```

Comment:
```suggestion
Policy("auth-provenance", repositoryid, "") :- check_passed(repositoryid, "mcn_provenance_level_three_1").
```

Reply:
Fixed in https://github.com/oracle-samples/macaron/pull/46/commits/73c1c81b6bdb6c999112532dab89103685eca48b